### PR TITLE
BREAKING CHANGE: migrate to webpack

### PR DESCRIPTION
**Pull request by saschanaz**
IMO mostly done, the last problem is that the highlighter is again randomly failing. 😱

### CODE REVIEW

**Comment by saschanaz** [comment]
>We still have requireJS. Is this something we will remove in next PRs?

This still uses requirejs but only to keep using `examples/basic.html` and some tests. I think @marcoscaceres preferred this non-optimized way when debugging because of sourcemap glitches on several browsers.

>Why are we importing styles as raw strings and not with style-loader? I was hoping we could minify the CSS in the production build.

`style-loader` will block future migration to native ES modules. Maybe a topic for new PRs.

File: package.json [hunk 1]
```diff
@@ -95,7 +95,7 @@
     "loading-indicator": "^2.0.0",
     "marked": "^0.5.2",
     "prompt": "^1.0.0",
-    "puppeteer": "^1.11.0",
+    "puppeteer-firefox": "^0.4.2",
```
[the remark this answers is not in the document]
Comment (saschanaz) [reply package.json]: Really wonder why the failure happens. It locally works and I have no way to debug it 🤯

**Comment by marcoscaceres** [comment]
Marked as breaking change, as the text loader will break downstream (that's ok tho 👍 ). It seems that once this is in place, it should be quite simple for downstream folks to fix their imports.

**Comment by saschanaz** [comment]
#1985 does help 🎉

**Pull request merged**